### PR TITLE
Add HTTP Headers option

### LINKED ISSUE #6784: [Feature] Setting Request Headers like Authorization Headers
**Is your feature request related to a problem? Please describe.**
I'm looking to subscribe to GitHub repositories as feeds. I am required to set the following headers
- `X-GitHub-Api-Version: 2022-11-28`
- `Accept: application/atom+xml`
- `Authorization: Bearer <TOKEN>`

However, I do not see the ability to set headers when adding a feed. Any help would be appreciated.
image

**Describe the solution you’d like**
I would love to have built-in fields to add headers when making GET requests to feeds.

**Describe alternatives you’ve considered**
I've considered deploying a proxy in between to add those headers but it would be great if it is part of the FreshRSS application.

### PR DESCRIPTION
**Pull request by ev-gor**
Closes #6784 

Changes proposed in this pull request:

- 'HTTP Headers' field was added to 'Add' and 'Update' subscription forms
- Values from this field are saved by feed/subscription controller
- A title and a hint for the field were added to lang files

How to test the feature manually:

1. Open 'Add' form for a new feed or 'Update' form for an existing one.
2. Add your HTTP headers to 'HTTP headers' field.
3. Save the form and update the feed.

Pull request checklist:

- [ ] clear commit messages
- [ ] code manually tested

### CODE REVIEW

File: app/Controllers/feedController.php [hunk 1]
```diff
@@ -178,6 +178,7 @@ public function addAction(): void {
 			$proxy_type = Minz_Request::paramString('proxy_type');
 			$request_method = Minz_Request::paramString('curl_method');
 			$request_fields = Minz_Request::paramString('curl_fields', true);
+			$headers = preg_split('/\r\n|\r|\n/', Minz_Request::paramString(key: 'http_headers'));
```
Comment (Alkarex) [inline app/Controllers/feedController.php]:
```suggestion
			$headers = Minz_Request::paramTextToArray('http_headers');
```

Please try that everywhere

**Comment by Alkarex** [comment]
There is an encoding issue to fix. Looking at it

**Comment by Alkarex** [comment]
Let me address the encoding issue in another PR, as it affects a few other fields as well.
Thanks, and please add a line for you in https://github.com/FreshRSS/FreshRSS/blob/edge/CREDITS.md

**Comment by Alkarex** [comment]
@ev-gor Could you please test https://github.com/FreshRSS/FreshRSS/pull/6821 with special characters such as `<'&">` ?

**Comment by ev-gor** [comment]
@Alkarex Yes, it works.

Before a server received a header:
`authorization: 'Bearer &lt;TOKEN&gt;&amp;'some'&amp;&quot;other&quot;'`

Now:
`authorization: 'Bearer <Token>&'some'&"other"'`

**Comment by Alkarex** [comment]
Excellent. Thanks for the feedback. Additional tests on all the other fields also welcome.

**Comment by Nikkiiw** [comment]
how to add a referer？ 
thx！

**Comment by Frenzie** [comment]
Under advanced, like this:

image

> Referer: https://example.com/page?q=123

(The typo-looking spelling of referer is intentional)

**Comment by Alkarex** [comment]
We could add a Referer example in the help box underneath. PR welcome

**Pull request merged**